java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -238,120 +119,16 @@ public void addIssue(int line, JavaCheck javaCheck, String message, @Nullable Do
-      return sourceFile.getKey();
-    }
-
-    @CheckForNull
-    private RuleKey getRuleKey(JavaCheck check) {
-      if (sonarComponents != null) {
-        for (Checks<JavaCheck> sonarChecks : sonarComponents.checks()) {
-          RuleKey ruleKey = sonarChecks.ruleKey(check);
-          if (ruleKey != null) {
-            return ruleKey;
-          }
-        }
-      }
-      return null;
-    }
-
-    @Override
-    public void addIssue(File file, JavaCheck check, int line, String message) {
-      RuleKey key = getRuleKey(check);
-      if (key != null) {
-        Issuable issuable = sonarComponents.issuableFor(file);
-        if (issuable != null) {
-          Issuable.IssueBuilder issueBuilder = issuable.newIssueBuilder()
-            .ruleKey(key)
-            .message(message);
-          if (line > -1) {
-            issueBuilder.line(line);
-          }
-          Issue issue = issueBuilder.build();
-          issuable.addIssue(issue);
-        }
-      }
-    }
-
-    /**
-     *
-     * DO NOT GO ON RELEASE WITH THIS CONSTANT SET TO TRUE
-     *
-     * **/
-    private static final boolean ENABLE_NEW_APIS = false;
-
-    @Override
-    public void reportIssue(JavaCheck javaCheck, Tree tree, String message) {
-      reportIssue(javaCheck, tree, message, ImmutableList.<Location>of(), null);
-    }
-
-    @Override
     public void reportIssue(JavaCheck javaCheck, Tree syntaxNode, String message, List<Location> secondary, @Nullable Integer cost) {
-      if (ENABLE_NEW_APIS) {
-        JavaCheckMessage checkMessage = new JavaCheckMessage(javaCheck, message);
-        AnalyzerMessage.TextSpan textSpan = textSpanFor(syntaxNode);
-        if (cost != null) {
-          checkMessage.setCost(cost);
-        }
-        checkMessage.setLine(textSpan.startLine);
-        AnalyzerMessage analyzerMessage = new AnalyzerMessage(javaCheck, file, textSpan, message, cost != null ? cost : 0);
-        for (Location location : secondary) {
-          AnalyzerMessage secondaryLocation = new AnalyzerMessage(javaCheck, file, textSpanFor(location.syntaxNode), location.msg, 0);
-          analyzerMessage.secondaryLocations.add(secondaryLocation);
-        }
-        checkMessage.setAnalyzerMessage(analyzerMessage);
-        sourceFile.log(checkMessage);
-      } else {
-        addIssue(syntaxNode, javaCheck, message, cost != null ? (double) cost : null);
+      File file = getFile();
+      AnalyzerMessage analyzerMessage = new AnalyzerMessage(javaCheck, file, textSpanFor(syntaxNode), message, cost != null ? cost : 0);
+      for (Location location : secondary) {
+        AnalyzerMessage secondaryLocation = new AnalyzerMessage(javaCheck, file, textSpanFor(location.syntaxNode), location.msg, 0);
+        analyzerMessage.secondaryLocations.add(secondaryLocation);
       }
+      issues.add(analyzerMessage);
+      addIssueForCheckMessageVerifier(analyzerMessage.getLine(), javaCheck, message, (double) (cost != null ? cost : 0));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Method 'TestJavaFileScannerContext.addIssueForCheckMessageVerifier(...)' is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)